server/src/modules/users/application/use-cases/authorize-user/__tests__/authorize-user-controller.test.unit.ts

```diff
@@ -0,0 +1,80 @@
+import express from 'express'
+import httpMocks from 'node-mocks-http'
+import { AppError } from '../../../../../../shared/core/app-error'
+import { Result } from '../../../../../../shared/core/result'
+import { AuthorizeUserDTO } from '../authorize-user-dto'
+import { AuthorizeUserErrors } from '../authorize-user-errors'
+import { AuthorizeUserUseCase } from '../authorize-user-use-case'
+import { AuthorizeUserController } from '../authorize-user-controller'
+import { mocks } from '../../../../../../test-utils'
+import { ParamList, ParamPair } from '../../../../../../shared/app/param-list'
+
+jest.mock('../authorize-user-use-case')
+jest.mock('../authorize-user-use-case')
```

Comment:
Seeing double

```suggestion
jest.mock('../authorize-user-use-case')
```

Reply:
BTW I've noticed that `jest.mock` doesn't actually need to be called if we use `jest.spyOn` on the object rather than on its class prototype.

i.e. here, you can set up a variable `authorizeUserUseCase = authorizeUser.authorizeUserUseCase`, then call `jest.spyOn(authorizeUserUseCase, 'execute')` instead of `jest.spyOn(AuthorizeUserUseCase.prototype, 'execute')`.